nil/common/concurrent/utils_test.go

```diff
@@ -0,0 +1,90 @@
+package concurrent
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+	"testing/synctest"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRunWithTimeout_AllFunctionsSucceed(t *testing.T) {
+	t.Parallel()
+
+	err := RunWithTimeout(
+		t.Context(),
+		1*time.Second,
+		WithSource(func(ctx context.Context) error {
+			return nil
+		}),
+		WithSource(func(ctx context.Context) error {
+			return nil
+		}))
+	require.NoError(t, err)
+}
+
+var errTest = errors.New("test error")
+
+func TestRunWithTimeout_FunctionFails(t *testing.T) {
+	t.Parallel()
+	synctest.Run(func() {
+		err := RunWithTimeout(
+			t.Context(),
+			1*time.Second,
+			WithSource(func(ctx context.Context) error {
+				return errTest
+			}),
+			WithSource(func(ctx context.Context) error {
+				time.Sleep(50 * time.Millisecond)
+				synctest.Wait()
+				if ctx.Err() != nil {
```

Comment:
Can be simplified a bit
```suggestion
				return ctx.Err()
```

You also need to fix test in CI and linter